### DISCUSSION TITLE
Actividad 1 - GRUPO 6

### DISCUSSION THREAD

**Discussion by GonzaloEBaez** [frlp-utn-ingsoft/catedra#94] [Show and tell]
https://github.com/GonzaloEBaez/vetsoft.git

**Comment by ivanduva** [comment]
Buenas,
La actividad tiene algunos problemas menores que les pedimos que solucionen para aprobar:

- El form de editar medicamento se rompe al ingresar un float en el campo dosis, en lugar de hacer la validación
- El form de producto permite cualquier caracter en precio y el campo Tipo no tiene mensaje de error en la validación.
- En el form de veterinario el campo telefono permite ingresar cualquier caracter cuando deberían ser solo números

Tienen tiempo para solucionarlo hasta la fecha de entrega de la próxima actividad de test que la informaremos la próxima clase.

Además les dejamos comentarios para las próximas actividades que deberían trabajar como grupo:

- Hay un test e2e que falla. Ahora no les pedimos solucionarlo porque esta actividad no era sobre tests, pero en la próxima lo van a tener que arreglar.
- Vimos que en varios casos hicieron las migraciones casi al finalizar la funcionalidad. Tengan cuidado con esto porque eso significa que en los commits anteriores la app no funciona. Recuerden hacer las migraciones cada vez que modifican el modelo y de probar la app cada vez que modifican algo para comprobar que funciona.
- En algunos casos vimos mensajes de commits haciendo referencia a archivos o funciones. Esto es redundante porque el commit ya tiene esa información. El mensaje debería contener una síntesis de la funcionalidad que resuelve el commit.
- También vimos algunos commits con mensajes duplicados al anterior. Por ejemplo: https://github.com/GonzaloEBaez/vetsoft/commit/7d31f76bc0ad50dbe6360524f59f78609cfe2e86. Puede haber pasado porque tuvieron que solucionar conflictos en un merge y copiaron el mensaje anterior. En ese caso les recomendamos usar el mensaje default de merge de git. Si fue intencional, es muy confuso. Eviten duplicar mensajes.
- Hay algunos commits que no son atómicos. Por ejemplo: https://github.com/GonzaloEBaez/vetsoft/commit/d7ebb23a6b94d8fba6afd9da781ae72a058afb30
- Los 3 comentarios anteriores tal vez puedan solucionarse si se animan como grupo a estandarizar de alguna manera los commits.

Saludos!

**Reply by ivanduva** [reply]
Buenas!
No vemos entrega de la corrección. Revisamos el proyecto y todavía existe el problema del teléfono.
Lo consideramos como una entrega fuera de término porque la fecha límite era la misma fecha que la Actividad 3 (26/05).
Recuerden que para la promoción tienen hasta 1 entrega fuera de término, como explicamos en el documento de Condiciones de Aprobación. A partir de 2 irían solo por la cursada. 
Pueden hacer la corrección que falta hasta la fecha de la próxima entrega (Miércoles 05/06).
Saludos

**Reply by GonzaloEBaez** [reply]
Buenas noches.

Dejamos el link con las correcciones pertinentes a lo charlado.

https://github.com/GonzaloEBaez/vetsoft.git

**Comment by ivanduva** [comment]
Buenas,
Los problemas empeoraron. En la lista de veterinarios el teléfono queda con valor `<django.db.models.fields.IntegerField>` y la app explota al guardar. El problema no es tanto que existan bugs sino que no se pruebe la app. Porque si se hubiera probado, inmediatamente se dan cuenta que el formulario no funciona. También notamos que el arreglo lo hizo otro integrante y no la que estaba encargada de ese issue.
Arreglen eso pero que lo haga Candela y los demás prueben la aplicación antes de mergear el PR.

**La última fecha límite de recuperatorio, tanto para esta como las demás actividades, será el último día de cursada antes de las vacaciones: Lunes 8 de Julio 23:59 hs.**

Saludos

**Reply by CandelAbregu** [reply]
Buenas noches Ivan , te envio este mensaje para comentarte que realice las correcciones correspondientes.
https://github.com/GonzaloEBaez/vetsoft

**Comment by ivanduva** [comment]
Ok, ahora lo vemos bien. Queda aprobado.
Por otro lado para las correcciones, van a tener que sacar siempre una release porque ahora el código queda desactualizado en el deploy. Así que saca una release con esto.

Saludos
